### PR TITLE
Support polymer-css-build

### LINKED ISSUE #202: Add support for polymer-css-build
polymer-css-build requires some coordination in ShadyCSS to avoid unnecessary work in the native shadow case, and to not double scope in the shadydom case.

### PR DESCRIPTION
**Pull request by azakus**
Fixes #202

### CODE REVIEW

File: src/style-util.js [hunk 1]
```diff
@@ -322,4 +314,56 @@ export function splitSelectorList(selector) {
...
+    return elementHasBuiltCss(/** @type {!HTMLElement} */(node));
+  }
+  return false;
+}
+
+/**
+ * @param {!HTMLElement} element
+ * @return {boolean}
+ */
+export function elementHasTargetedBuild(element) {
```
[the remark this answers is not in the document]
Comment (azakus) [reply src/style-util.js]: Doesn't seem to be used, removing.

File: src/document-watcher.js [hunk 2]
```diff
@@ -124,7 +124,7 @@ function handler(mxns) {
       let root = n.getRootNode();
       let currentScope = getCurrentScope(n);
       // node was scoped, but now is in document
-      if (currentScope && root === n.ownerDocument) {
+      if (currentScope && root === n.ownerDocument && !isNodeWithBuiltCss(n)) {
```
[the remark this answers is not in the document]
Comment (azakus) [reply src/document-watcher.js]: Done

File: entrypoints/apply-shim.js [hunk 3]
```diff
@@ -51,6 +51,9 @@ class ApplyShimInterface {
    */
   prepareTemplate(template, elementName) {
     this.ensure();
+    if (elementHasBuiltCss(template)) {
```
[the remark this answers is not in the document]
Comment (azakus) [reply entrypoints/apply-shim.js]: The best spot is probably in `StyleUtil::elementHasBuiltCss`

File: src/scoping-shim.js [hunk 4]
```diff
@@ -255,6 +253,10 @@ export default class ScopingShim {
       if (!template && !this._isRootOwner(host)) {
         return;
       }
+      // bail early if the template was built with polymer-css-build
```
[the remark this answers is not in the document]
Comment (azakus) [reply src/scoping-shim.js]: Done

File: src/style-util.js [hunk 5]
```diff
@@ -326,7 +326,11 @@ const CSS_BUILD_ATTR = 'css-build';
  */
 export function getCssBuild(element) {
   if (element.__cssBuild === undefined) {
-    element.__cssBuild = element.getAttribute(CSS_BUILD_ATTR);
+    const buildComment = getBuildComment(element);
```
Comment (sorvell) [inline src/style-util.js]: let's only do this work if we don't have an attribute value.

Comment (sorvell) [reply src/style-util.js]: Comment mode needs a test.

Comment (azakus) [reply src/style-util.js]: Done and done

**Pull request merged**